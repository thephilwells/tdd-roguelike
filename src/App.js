import React, { createContext } from 'react'
import { render, Text } from 'ink'
import { Arena } from './components/Arena'

export const GameContext = createContext()

export const App = () => {
  return <Arena />
}

const defaultState = {
  title: 'TDD Roguelike!'
}

render(
  <GameContext.Provider value={defaultState}>
    <App />
  </GameContext.Provider>,
  {patchConsole: false}
)
