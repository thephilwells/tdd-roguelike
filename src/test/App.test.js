import React from "react";
import { render } from "ink-testing-library";
import { App, GameContext } from "../App";

describe("APP", () => {
  it("returns the name of the game", () => {
    // Arrange
    const { lastFrame } = render(
      <GameContext.Provider value={{ title: "FOO!" }}>
        <App />
      </GameContext.Provider>
    );

    // Assert
    expect(lastFrame()).toMatchSnapshot();
  });
});
