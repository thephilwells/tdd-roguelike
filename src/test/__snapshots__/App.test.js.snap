// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP returns the name of the game 1`] = `
"╔══════════════════════════════════════════════════════════════════════════════╗
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                     FOO!                                     ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
║                         [Press any key to start...]                          ║
║                                                                              ║
║                                                                              ║
║                                                                              ║
╚══════════════════════════════════════════════════════════════════════════════╝"
`;
